### ISSUE TITLE
format method gives wrong hour, minite and second

### ISSUE THREAD

**Issue by git-pwn** [moment/moment#5681]
**Describe the bug**

The ` HH:mm:ss` format gives wrong date string under certain circumstances

**To Reproduce**

```js
moment(new Date('2020-08-11 23:59:59')).format('yyyy-MM-DD HH:mm:ss')
// output: "2020-08-11 00:00:00" BUGGY
moment(new Date('2020-08-11 23:59:59')).format('HH:mm:ss')
// output: "23:59:59"
moment(new Date('2020-08-11 23:59:59')).format('HH:mm:ss yyyy-MM-DD')
// output: "23:59:59 2020-08-11"
```

**Expected behavior**
-  `yyyy-MM-DD HH:mm:ss` should gives the correct value
- formatted display should be consistent regardless of the order of the format string

**Additional context**

moment.js VERSION:  2.27.0

**Comment by MinusFour** [comment]
This is a duplicate of #5571, it's getting fixed soon apparently.

**Comment by marwahaha** [comment]
fixed in 2.28.0
